resources/osl_mysql_test.rb

```diff
@@ -0,0 +1,38 @@
+provides :osl_mysql_test
+resource_name :osl_mysql_test
+unified_mode true
+
+default_action :create
+
+property :database, String, name_property: true
+property :username, String, required: true
+property :password, String, required: true
+property :server_password, String, default: 'osl_mysql_test'
+property :encoding, String, default: 'utf8mb4'
+property :collation, String, default: 'utf8mb4_unicode_ci'
+property :database_parameters, Hash, default: {}
```

Comment:
I believe you can simplify this to the following:
```suggestion
property :database_parameters, Hash
```

Reply:
Unfortunately, this will not work, as the default is nulltype instead of a Hash, so there is an error while running the resource, when we try to loop through the hash, when there is none given.